components/controls/Boolean.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="flex gap-3 cursor-pointer" @click="onToggle">
+    <div
+      class="text-primary w-6 h-6 border border-black flex items-center transition-all"
+      :class="{ 'bg-primary-light border-primary': model }"
+    >
+      <Icon v-if="model" class="text-primary w-8 h-8" name="check" />
+    </div>
+    <div>{{ label }}</div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import Icon from '~/components/general/Icon.vue'
+import { useControls } from '~/composables/controls'
+
+const props = withDefaults(
+  defineProps<{
+    keyValue: string
```

Comment:
Since this emulates an `<input>` component, this might me more fitting
```suggestion
    name: string
```